catalog/app/utils/AWS/S3.js

```diff
@@ -83,26 +83,23 @@ function useSmartS3() {
           )
           req.on('sign', () => {
             if (req.httpRequest[PRESIGN]) return
+
             // Monkey-patch the request object after it has been signed and save the original
             // values in case of retry.
+            const origEndpoint = req.httpRequest.endpoint
+            const origPath = req.httpRequest.path
+
             req.httpRequest[PROXIED] = {
-              endpoint: req.httpRequest.endpoint,
-              path: req.httpRequest.path,
+              endpoint: origEndpoint,
+              path: origPath,
             }
             const basePath = endpoint.path.replace(/\/$/, '')
-            // handle buckets with dots in their names
-            if (
-              req.httpRequest.path.startsWith(`/${b}`) &&
-              !req.httpRequest.endpoint.host.startsWith(`${b}.`)
-            ) {
-              req.httpRequest.path = req.httpRequest.path.replace(`/${b}`, '')
-            }
 
             req.httpRequest.endpoint = endpoint
             req.httpRequest.path =
               type === 'select'
-                ? `${basePath}${req.httpRequest.path}`
-                : `${basePath}/${req.httpRequest.region}/${b}${req.httpRequest.path}`
+                ? `${basePath}${origPath}`
```

Comment:
This likely breaks S3 select - but I can't remember where we use it, so don't know how to test it.

Also, [this hostname](https://github.com/quiltdata/quilt/blob/master/lambdas/s3select/src/t4_lambda_s3select/__init__.py#L37) can't possibly work with bucket names that contain dots, right?.. We need path-style access here.

Reply:
> This likely breaks S3 select - but I can't remember where we use it

From memory we use S3 select in two places: package-select lambda, es/indexer to pull package-level metadata.

Try searching for `select_object_content` (the boto3 call).
I believe that is only dispatched from `query_manifest_content` in utils.py.

Reply:
This is a lambda called from JS, though. But anyways, figured it out: it's only used to get object access counts when you're not logged in. So https://open.quiltdata.com/ is pretty much the only place this comes up. Bucket names with dots are broken in that case - but we've never added such buckets to the registry of open data, so we wouldn't know.

So turns out my change is actually ok here. I tested it on the registry of open data, and it still works fine (on buckets without dots, that is). Buckets with dots are broken, but they never worked in the first place: the s3select lambda just returns a 500. I'll see if it's an easy fix - but it shouldn't block this PR.

Reply:
Well, s3select was not as trivial of a fix as I thought... But, turns out, we only use s3select for the _analytics bucket_ - so we really don't need to worry about dots there.